<Activity mc:Ignorable="sap sap2010" x:Class="ProcessTransaction" xmlns="http://schemas.microsoft.com/netfx/2009/xaml/activities" xmlns:av="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" xmlns:mva="clr-namespace:Microsoft.VisualBasic.Activities;assembly=System.Activities" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:sap="http://schemas.microsoft.com/netfx/2009/xaml/activities/presentation" xmlns:sap2010="http://schemas.microsoft.com/netfx/2010/xaml/activities/presentation" xmlns:scg="clr-namespace:System.Collections.Generic;assembly=mscorlib" xmlns:sco="clr-namespace:System.Collections.ObjectModel;assembly=mscorlib" xmlns:sd="clr-namespace:System.Data;assembly=System.Data" xmlns:snm="clr-namespace:System.Net.Mail;assembly=System" xmlns:ui="http://schemas.uipath.com/workflow/activities" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
  <x:Members>
    <x:Property Name="in_TransactionItem" Type="InArgument(snm:MailMessage)" />
    <x:Property Name="in_Config" Type="InArgument(scg:Dictionary(x:String, x:Object))" />
  </x:Members>
  <mva:VisualBasic.Settings>
    <x:Null />
  </mva:VisualBasic.Settings>
  <sap:VirtualizedContainerService.HintSize>1085,2276</sap:VirtualizedContainerService.HintSize>
  <sap2010:WorkflowViewState.IdRef>ProcessTransaction_1</sap2010:WorkflowViewState.IdRef>
  <TextExpression.NamespacesForImplementation>
    <sco:Collection x:TypeArguments="x:String">
      <x:String>System.Activities</x:String>
      <x:String>System.Activities.Statements</x:String>
      <x:String>System.Activities.Expressions</x:String>
      <x:String>System.Activities.Validation</x:String>
      <x:String>System.Activities.XamlIntegration</x:String>
      <x:String>Microsoft.VisualBasic</x:String>
      <x:String>Microsoft.VisualBasic.Activities</x:String>
      <x:String>System</x:String>
      <x:String>System.Collections</x:String>
      <x:String>System.Collections.Generic</x:String>
      <x:String>System.Data</x:String>
      <x:String>System.Diagnostics</x:String>
      <x:String>System.Drawing</x:String>
      <x:String>System.IO</x:String>
      <x:String>System.Linq</x:String>
      <x:String>System.Net.Mail</x:String>
      <x:String>System.Xml</x:String>
      <x:String>System.Xml.Linq</x:String>
      <x:String>UiPath.Core</x:String>
      <x:String>UiPath.Core.Activities</x:String>
      <x:String>System.Windows.Markup</x:String>
      <x:String>System.Xml.Serialization</x:String>
    </sco:Collection>
  </TextExpression.NamespacesForImplementation>
  <TextExpression.ReferencesForImplementation>
    <sco:Collection x:TypeArguments="AssemblyReference">
      <AssemblyReference>System.Activities</AssemblyReference>
      <AssemblyReference>Microsoft.VisualBasic</AssemblyReference>
      <AssemblyReference>mscorlib</AssemblyReference>
      <AssemblyReference>System.Data</AssemblyReference>
      <AssemblyReference>System</AssemblyReference>
      <AssemblyReference>System.Drawing</AssemblyReference>
      <AssemblyReference>System.Core</AssemblyReference>
      <AssemblyReference>System.Xml</AssemblyReference>
      <AssemblyReference>System.Xml.Linq</AssemblyReference>
      <AssemblyReference>PresentationFramework</AssemblyReference>
      <AssemblyReference>WindowsBase</AssemblyReference>
      <AssemblyReference>PresentationCore</AssemblyReference>
      <AssemblyReference>System.Xaml</AssemblyReference>
      <AssemblyReference>System.ComponentModel.Composition</AssemblyReference>
      <AssemblyReference>System.ServiceModel</AssemblyReference>
      <AssemblyReference>Microsoft.VisualStudio.Services.Common</AssemblyReference>
      <AssemblyReference>UiPath.UiAutomation.Activities</AssemblyReference>
      <AssemblyReference>UiPath.System.Activities</AssemblyReference>
      <AssemblyReference>UiPath.Excel</AssemblyReference>
      <AssemblyReference>UiPath.Mail</AssemblyReference>
      <AssemblyReference>UiPath.Word</AssemblyReference>
      <AssemblyReference>System.ValueTuple</AssemblyReference>
      <AssemblyReference>System.Data.DataSetExtensions</AssemblyReference>
      <AssemblyReference>System.Runtime.WindowsRuntime</AssemblyReference>
    </sco:Collection>
  </TextExpression.ReferencesForImplementation>
  <Sequence sap2010:Annotation.AnnotationText="Description: In this file all other process specific files will be invoked. If an application exception occurs, the current transaction can be retried. If a BRE is thrown, the transaction will be skipped. Can be a flowchart or sequence. " DisplayName="ProcessTransaction" sap:VirtualizedContainerService.HintSize="1065,2216" sap2010:WorkflowViewState.IdRef="Sequence_1">
    <sap:WorkflowViewStateService.ViewState>
      <scg:Dictionary x:TypeArguments="x:String, x:Object">
        <x:Boolean x:Key="IsExpanded">True</x:Boolean>
        <x:Boolean x:Key="IsAnnotationDocked">True</x:Boolean>
      </scg:Dictionary>
    </sap:WorkflowViewStateService.ViewState>
    <Flowchart DisplayName="Process-Layer" sap:VirtualizedContainerService.HintSize="1023,2082" sap2010:WorkflowViewState.IdRef="Flowchart_1">
      <Flowchart.Variables>
        <Variable x:TypeArguments="s:Exception" Name="ProcessApplicationException" />
        <Variable x:TypeArguments="ui:BusinessRuleException" Name="ProcessBusinessException" />
        <Variable x:TypeArguments="sd:DataTable" Default="[New DataTable()]" Name="dtNLPDataReviews" />
        <Variable x:TypeArguments="x:String" Name="strProductName" />
        <Variable x:TypeArguments="scg:Dictionary(x:String, x:Int32)" Name="dictAmazonProductReview" />
        <Variable x:TypeArguments="scg:Dictionary(x:String, x:String)" Name="dictFlipkartProductReview" />
        <Variable x:TypeArguments="scg:Dictionary(x:String, x:String)" Name="dictGoogleProductReview" />
      </Flowchart.Variables>
      <sap:WorkflowViewStateService.ViewState>
        <scg:Dictionary x:TypeArguments="x:String, x:Object">
          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
          <x:Boolean x:Key="IsPinned">False</x:Boolean>
          <av:Point x:Key="ShapeLocation">270,2.5</av:Point>
          <av:Size x:Key="ShapeSize">60,75</av:Size>
          <av:PointCollection x:Key="ConnectorLocation">300,77.5 300,121</av:PointCollection>
          <x:Double x:Key="Width">989</x:Double>
          <x:Double x:Key="Height">2045.5</x:Double>
        </scg:Dictionary>
      </sap:WorkflowViewStateService.ViewState>
      <Flowchart.StartNode>
        <x:Reference>__ReferenceID29</x:Reference>
      </Flowchart.StartNode>
      <FlowStep x:Name="__ReferenceID14" sap2010:WorkflowViewState.IdRef="FlowStep_1">
        <sap:WorkflowViewStateService.ViewState>
          <scg:Dictionary x:TypeArguments="x:String, x:Object">
            <av:Point x:Key="ShapeLocation">200,454.5</av:Point>
            <av:Size x:Key="ShapeSize">200,78</av:Size>
            <av:PointCollection x:Key="ConnectorLocation">300,532.5 300,576.5</av:PointCollection>
          </scg:Dictionary>
        </sap:WorkflowViewStateService.ViewState>
        <TryCatch sap2010:Annotation.AnnotationText="AE/BE Handling Necessary" DisplayName="TryCatch SubProcess #2 Amazon Product Details" sap:VirtualizedContainerService.HintSize="200,78" sap2010:WorkflowViewState.IdRef="TryCatch_1">
          <sap:WorkflowViewStateService.ViewState>
            <scg:Dictionary x:TypeArguments="x:String, x:Object">
              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
              <x:Boolean x:Key="IsAnnotationDocked">True</x:Boolean>
            </scg:Dictionary>
          </sap:WorkflowViewStateService.ViewState>
          <TryCatch.Try>
            <Sequence DisplayName="Invoke Sub-Process #1 And Set Progress" sap:VirtualizedContainerService.HintSize="421,179" sap2010:WorkflowViewState.IdRef="Sequence_7">
              <sap:WorkflowViewStateService.ViewState>
                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                  <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                </scg:Dictionary>
              </sap:WorkflowViewStateService.ViewState>
              <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke Sub-Process #1 CPA - Get Product Details From Amazon" sap:VirtualizedContainerService.HintSize="379,87" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_21" UnSafe="False" WorkflowFileName="Sub-Processes\URNP - Get Product Reviewa From Amazon.xaml">
                <ui:InvokeWorkflowFile.Arguments>
                  <InArgument x:TypeArguments="x:String" x:Key="in_StrAmazonURL">[in_Config("Amazon_URL").ToString]</InArgument>
                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutS">[Int32.Parse(in_Config("TimeoutS").ToString)]</InArgument>
                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutM">[Int32.Parse(in_Config("TimeoutM").ToString)]</InArgument>
                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutL">[Int32.Parse(in_Config("TimeoutL").ToString)]</InArgument>
                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntDefaultDelay">[Int32.Parse(in_Config("Default_Delay").ToString)]</InArgument>
                  <InArgument x:TypeArguments="x:String" x:Key="in_StrProductName">[strProductName]</InArgument>
                  <InArgument x:TypeArguments="x:String" x:Key="in_StrBrowserType">[in_Config("Browser_Chrome").ToString]</InArgument>
                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntNoOfReviewsToProcess">[Int32.Parse(in_Config("ReviewsToProcess").ToString)]</InArgument>
                  <InOutArgument x:TypeArguments="sd:DataTable" x:Key="io_DtNLPReviews">[dtNLPDataReviews]</InOutArgument>
                  <OutArgument x:TypeArguments="scg:Dictionary(x:String, x:Int32)" x:Key="out_DictAmazonProductReviews">[dictAmazonProductReview]</OutArgument>
                </ui:InvokeWorkflowFile.Arguments>
              </ui:InvokeWorkflowFile>
            </Sequence>
          </TryCatch.Try>
          <TryCatch.Catches>
            <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="425,21" sap2010:WorkflowViewState.IdRef="Catch`1_1">
              <sap:WorkflowViewStateService.ViewState>
                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                  <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                </scg:Dictionary>
              </sap:WorkflowViewStateService.ViewState>
              <ActivityAction x:TypeArguments="s:Exception">
                <ActivityAction.Argument>
                  <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                </ActivityAction.Argument>
                <Assign DisplayName="Assign ProcessApplicationException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_26">
                  <Assign.To>
                    <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                  </Assign.To>
                  <Assign.Value>
                    <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                  </Assign.Value>
                </Assign>
              </ActivityAction>
            </Catch>
            <Catch x:TypeArguments="ui:BusinessRuleException" sap:VirtualizedContainerService.HintSize="425,21" sap2010:WorkflowViewState.IdRef="Catch`1_2">
              <sap:WorkflowViewStateService.ViewState>
                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                  <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                </scg:Dictionary>
              </sap:WorkflowViewStateService.ViewState>
              <ActivityAction x:TypeArguments="ui:BusinessRuleException">
                <ActivityAction.Argument>
                  <DelegateInArgument x:TypeArguments="ui:BusinessRuleException" Name="exception" />
                </ActivityAction.Argument>
                <Assign DisplayName="Assign ProcessBusinessException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_9">
                  <Assign.To>
                    <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                  </Assign.To>
                  <Assign.Value>
                    <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                  </Assign.Value>
                </Assign>
              </ActivityAction>
            </Catch>
          </TryCatch.Catches>
        </TryCatch>
        <FlowStep.Next>
          <FlowDecision x:Name="__ReferenceID0" Condition="[IsNothing(ProcessApplicationException) AND IsNothing(ProcessBusinessException)]" DisplayName="AE/BE Success?" sap:VirtualizedContainerService.HintSize="87,87" sap2010:WorkflowViewState.IdRef="FlowDecision_1">
            <sap:WorkflowViewStateService.ViewState>
              <scg:Dictionary x:TypeArguments="x:String, x:Object">
                <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                <av:Point x:Key="ShapeLocation">256.5,576.5</av:Point>
                <av:Size x:Key="ShapeSize">87,87</av:Size>
                <av:PointCollection x:Key="FalseConnector">343.5,620 373.5,620 373.5,576.5 720,576.5 720,606.5</av:PointCollection>
                <av:PointCollection x:Key="TrueConnector">256.5,620 226.5,620 226.5,724.5 310,724.5 310,754.5</av:PointCollection>
              </scg:Dictionary>
            </sap:WorkflowViewStateService.ViewState>
            <FlowDecision.True>
              <FlowStep x:Name="__ReferenceID3" sap2010:WorkflowViewState.IdRef="FlowStep_4">
                <sap:WorkflowViewStateService.ViewState>
                  <scg:Dictionary x:TypeArguments="x:String, x:Object">
                    <av:Point x:Key="ShapeLocation">210,754.5</av:Point>
                    <av:Size x:Key="ShapeSize">200,78</av:Size>
                    <av:PointCollection x:Key="ConnectorLocation">310,832.5 310,896.5</av:PointCollection>
                  </scg:Dictionary>
                </sap:WorkflowViewStateService.ViewState>
                <TryCatch sap2010:Annotation.AnnotationText="Only AE Handling Necessary" DisplayName="TryCatch SubProcess #3 Flipkart Product Details" sap:VirtualizedContainerService.HintSize="200,78" sap2010:WorkflowViewState.IdRef="TryCatch_3">
                  <sap:WorkflowViewStateService.ViewState>
                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                      <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                      <x:Boolean x:Key="IsAnnotationDocked">True</x:Boolean>
                    </scg:Dictionary>
                  </sap:WorkflowViewStateService.ViewState>
                  <TryCatch.Try>
                    <Sequence DisplayName="Invoke Sub-Process #2 And Set Transaction Progress" sap:VirtualizedContainerService.HintSize="421,179" sap2010:WorkflowViewState.IdRef="Sequence_14">
                      <sap:WorkflowViewStateService.ViewState>
                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                        </scg:Dictionary>
                      </sap:WorkflowViewStateService.ViewState>
                      <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke Sub-Process #2 CPA - Get Product Details From Flipkart" sap:VirtualizedContainerService.HintSize="379,87" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_3" UnSafe="False" WorkflowFileName="Sub-Processes\URNP - Get Product Reviews From Flipkart.xaml">
                        <ui:InvokeWorkflowFile.Arguments>
                          <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutS">[Int32.Parse(in_Config("TimeoutS").ToString)]</InArgument>
                          <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutM">[Int32.Parse(in_Config("TimeoutM").ToString)]</InArgument>
                          <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutL">[Int32.Parse(in_Config("TimeoutL").ToString)]</InArgument>
                          <InArgument x:TypeArguments="x:Int32" x:Key="in_IntDefaultDelay">[Int32.Parse(in_Config("Default_Delay").ToString)]</InArgument>
                          <InArgument x:TypeArguments="x:String" x:Key="in_StrProductName">[strProductName]</InArgument>
                          <InArgument x:TypeArguments="x:String" x:Key="in_StrBrowserType">[in_Config("Browser_IE").ToString]</InArgument>
                          <InArgument x:TypeArguments="x:Int32" x:Key="in_IntNoOfReviewsToProcess">[Int32.Parse(in_Config("ReviewsToProcess").ToString)]</InArgument>
                          <InArgument x:TypeArguments="x:String" x:Key="in_StrFlipkartURL">[in_Config("Flipkart_URL").ToString]</InArgument>
                          <InOutArgument x:TypeArguments="sd:DataTable" x:Key="io_DtNLPReviews">[dtNLPDataReviews]</InOutArgument>
                          <OutArgument x:TypeArguments="scg:Dictionary(x:String, x:String)" x:Key="out_DictFlipkartProductReviews">[dictFlipkartProductReview]</OutArgument>
                        </ui:InvokeWorkflowFile.Arguments>
                      </ui:InvokeWorkflowFile>
                    </Sequence>
                  </TryCatch.Try>
                  <TryCatch.Catches>
                    <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="425,21" sap2010:WorkflowViewState.IdRef="Catch`1_5">
                      <sap:WorkflowViewStateService.ViewState>
                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                          <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                          <x:Boolean x:Key="IsPinned">False</x:Boolean>
                        </scg:Dictionary>
                      </sap:WorkflowViewStateService.ViewState>
                      <ActivityAction x:TypeArguments="s:Exception">
                        <ActivityAction.Argument>
                          <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                        </ActivityAction.Argument>
                        <Assign DisplayName="Assign ProcessApplicationException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_30">
                          <Assign.To>
                            <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                          </Assign.To>
                          <Assign.Value>
                            <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                          </Assign.Value>
                        </Assign>
                      </ActivityAction>
                    </Catch>
                    <Catch x:TypeArguments="ui:BusinessRuleException" sap:VirtualizedContainerService.HintSize="425,21" sap2010:WorkflowViewState.IdRef="Catch`1_10">
                      <sap:WorkflowViewStateService.ViewState>
                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                          <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                          <x:Boolean x:Key="IsPinned">False</x:Boolean>
                        </scg:Dictionary>
                      </sap:WorkflowViewStateService.ViewState>
                      <ActivityAction x:TypeArguments="ui:BusinessRuleException">
                        <ActivityAction.Argument>
                          <DelegateInArgument x:TypeArguments="ui:BusinessRuleException" Name="exception" />
                        </ActivityAction.Argument>
                        <Sequence DisplayName="Setup New Exception (Amalgamating Message/Source) Then Throw" sap:VirtualizedContainerService.HintSize="304,211" sap2010:WorkflowViewState.IdRef="Sequence_18">
                          <sap:WorkflowViewStateService.ViewState>
                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                            </scg:Dictionary>
                          </sap:WorkflowViewStateService.ViewState>
                          <Assign DisplayName="Assign ProcessBusinessException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_31">
                            <Assign.To>
                              <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                            </Assign.To>
                            <Assign.Value>
                              <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                            </Assign.Value>
                          </Assign>
                          <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="262,22" sap2010:WorkflowViewState.IdRef="Throw_10" />
                        </Sequence>
                      </ActivityAction>
                    </Catch>
                  </TryCatch.Catches>
                </TryCatch>
                <FlowStep.Next>
                  <FlowDecision x:Name="__ReferenceID2" Condition="[IsNothing(ProcessApplicationException) AND IsNothing(ProcessBusinessException)]" DisplayName="AE/BE Success?" sap:VirtualizedContainerService.HintSize="87,87" sap2010:WorkflowViewState.IdRef="FlowDecision_2">
                    <sap:WorkflowViewStateService.ViewState>
                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                        <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                        <av:Point x:Key="ShapeLocation">266.5,896.5</av:Point>
                        <av:Size x:Key="ShapeSize">87,87</av:Size>
                        <av:PointCollection x:Key="FalseConnector">353.5,940 720,940 720,956.5</av:PointCollection>
                        <av:PointCollection x:Key="TrueConnector">266.5,940 250,940 250,1091 310,1091 310,1121</av:PointCollection>
                      </scg:Dictionary>
                    </sap:WorkflowViewStateService.ViewState>
                    <FlowDecision.True>
                      <FlowStep x:Name="__ReferenceID7" sap2010:WorkflowViewState.IdRef="FlowStep_8">
                        <sap:WorkflowViewStateService.ViewState>
                          <scg:Dictionary x:TypeArguments="x:String, x:Object">
                            <av:Point x:Key="ShapeLocation">210,1121</av:Point>
                            <av:Size x:Key="ShapeSize">200,78</av:Size>
                            <av:PointCollection x:Key="ConnectorLocation">310,1199 310,1229 318.5,1229 318.5,1249</av:PointCollection>
                          </scg:Dictionary>
                        </sap:WorkflowViewStateService.ViewState>
                        <TryCatch sap2010:Annotation.AnnotationText="No AE/BE Handling Required" DisplayName="TryCatch SubProcess #4 Google Product Details" sap:VirtualizedContainerService.HintSize="200,78" sap2010:WorkflowViewState.IdRef="TryCatch_7">
                          <sap:WorkflowViewStateService.ViewState>
                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                              <x:Boolean x:Key="IsAnnotationDocked">True</x:Boolean>
                            </scg:Dictionary>
                          </sap:WorkflowViewStateService.ViewState>
                          <TryCatch.Try>
                            <Sequence DisplayName="Invoke Sub-Process #3 Google Product Details" sap:VirtualizedContainerService.HintSize="421,179" sap2010:WorkflowViewState.IdRef="Sequence_15">
                              <sap:WorkflowViewStateService.ViewState>
                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                  <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                </scg:Dictionary>
                              </sap:WorkflowViewStateService.ViewState>
                              <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke Sub-Process #3" sap:VirtualizedContainerService.HintSize="379,87" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_19" UnSafe="False" WorkflowFileName="Sub-Processes\URNP - Get Product Reviewa From Google.xaml">
                                <ui:InvokeWorkflowFile.Arguments>
                                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutS">[Int32.Parse(in_Config("TimeoutS").ToString)]</InArgument>
                                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutM">[Int32.Parse(in_Config("TimeoutM").ToString)]</InArgument>
                                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntTimeoutL">[Int32.Parse(in_Config("TimeoutL").ToString)]</InArgument>
                                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntDefaultDelay">[Int32.Parse(in_Config("Default_Delay").ToString)]</InArgument>
                                  <InArgument x:TypeArguments="x:String" x:Key="in_StrProductName">[strProductName]</InArgument>
                                  <InArgument x:TypeArguments="x:String" x:Key="in_StrBrowserType">[in_Config("Browser_Chrome").ToString]</InArgument>
                                  <InArgument x:TypeArguments="x:Int32" x:Key="in_IntNoOfReviewsToProcess">[Int32.Parse(in_Config("ReviewsToProcess").ToString)]</InArgument>
                                  <InArgument x:TypeArguments="x:String" x:Key="in_StrGoogleURL">[in_Config("Google_URL").ToString]</InArgument>
                                  <InOutArgument x:TypeArguments="sd:DataTable" x:Key="io_DtNLPReviews">[dtNLPDataReviews]</InOutArgument>
                                  <OutArgument x:TypeArguments="scg:Dictionary(x:String, x:String)" x:Key="out_DictGoogleProductReviews">[dictGoogleProductReview]</OutArgument>
                                </ui:InvokeWorkflowFile.Arguments>
                              </ui:InvokeWorkflowFile>
                            </Sequence>
                          </TryCatch.Try>
                          <TryCatch.Catches>
                            <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="425,21" sap2010:WorkflowViewState.IdRef="Catch`1_12">
                              <sap:WorkflowViewStateService.ViewState>
                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                  <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                </scg:Dictionary>
                              </sap:WorkflowViewStateService.ViewState>
                              <ActivityAction x:TypeArguments="s:Exception">
                                <ActivityAction.Argument>
                                  <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                </ActivityAction.Argument>
                                <Sequence DisplayName="Setup New Exception (Amalgamating Message/Source) Then Throw" sap:VirtualizedContainerService.HintSize="304,211" sap2010:WorkflowViewState.IdRef="Sequence_19">
                                  <sap:WorkflowViewStateService.ViewState>
                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                      <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                    </scg:Dictionary>
                                  </sap:WorkflowViewStateService.ViewState>
                                  <Assign DisplayName="Assign ProcessApplicationException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_34">
                                    <Assign.To>
                                      <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                                    </Assign.To>
                                    <Assign.Value>
                                      <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                                    </Assign.Value>
                                  </Assign>
                                  <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="262,22" sap2010:WorkflowViewState.IdRef="Throw_11" />
                                </Sequence>
                              </ActivityAction>
                            </Catch>
                            <Catch x:TypeArguments="ui:BusinessRuleException" sap:VirtualizedContainerService.HintSize="425,21" sap2010:WorkflowViewState.IdRef="Catch`1_13">
                              <sap:WorkflowViewStateService.ViewState>
                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                  <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                </scg:Dictionary>
                              </sap:WorkflowViewStateService.ViewState>
                              <ActivityAction x:TypeArguments="ui:BusinessRuleException">
                                <ActivityAction.Argument>
                                  <DelegateInArgument x:TypeArguments="ui:BusinessRuleException" Name="exception" />
                                </ActivityAction.Argument>
                                <Sequence DisplayName="Setup New Exception (Amalgamating Message/Source) Then Throw" sap:VirtualizedContainerService.HintSize="304,211" sap2010:WorkflowViewState.IdRef="Sequence_20">
                                  <sap:WorkflowViewStateService.ViewState>
                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                      <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                    </scg:Dictionary>
                                  </sap:WorkflowViewStateService.ViewState>
                                  <Assign DisplayName="Assign ProcessBusinessException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_35">
                                    <Assign.To>
                                      <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                                    </Assign.To>
                                    <Assign.Value>
                                      <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                                    </Assign.Value>
                                  </Assign>
                                  <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="262,22" sap2010:WorkflowViewState.IdRef="Throw_12" />
                                </Sequence>
                              </ActivityAction>
                            </Catch>
                          </TryCatch.Catches>
                        </TryCatch>
                        <FlowStep.Next>
                          <FlowDecision x:Name="__ReferenceID10" Condition="[IsNothing(ProcessApplicationException) AND IsNothing(ProcessBusinessException)]" DisplayName="AE/BE Success?" sap:VirtualizedContainerService.HintSize="87,87" sap2010:WorkflowViewState.IdRef="FlowDecision_5">
                            <sap:WorkflowViewStateService.ViewState>
                              <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                <av:Point x:Key="ShapeLocation">275,1249</av:Point>
                                <av:Size x:Key="ShapeSize">87,87</av:Size>
                                <av:PointCollection x:Key="FalseConnector">362,1292.5 392,1292.5 392,1246.5 730,1246.5 730,1276.5</av:PointCollection>
                                <av:PointCollection x:Key="TrueConnector">275,1292.5 250,1292.5 250,1391 310,1391 310,1421</av:PointCollection>
                              </scg:Dictionary>
                            </sap:WorkflowViewStateService.ViewState>
                            <FlowDecision.True>
                              <FlowStep x:Name="__ReferenceID23">
                                <sap:WorkflowViewStateService.ViewState>
                                  <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                    <av:Point x:Key="ShapeLocation">210,1421</av:Point>
                                    <av:Size x:Key="ShapeSize">200,78</av:Size>
                                    <av:PointCollection x:Key="ConnectorLocation">310,1499 310,1529 318.5,1529 318.5,1549</av:PointCollection>
                                  </scg:Dictionary>
                                </sap:WorkflowViewStateService.ViewState>
                                <TryCatch sap2010:Annotation.AnnotationText="No AE/BE Handling Required" DisplayName="TryCatch SubProcess #5 Extract NPL Results" sap:VirtualizedContainerService.HintSize="452,455" sap2010:WorkflowViewState.IdRef="TryCatch_15">
                                  <sap:WorkflowViewStateService.ViewState>
                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                      <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                      <x:Boolean x:Key="IsAnnotationDocked">True</x:Boolean>
                                    </scg:Dictionary>
                                  </sap:WorkflowViewStateService.ViewState>
                                  <TryCatch.Try>
                                    <Sequence DisplayName="Invoke Sub-Process #5 Create Report" sap:VirtualizedContainerService.HintSize="414,179" sap2010:WorkflowViewState.IdRef="Sequence_22">
                                      <sap:WorkflowViewStateService.ViewState>
                                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                          <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                        </scg:Dictionary>
                                      </sap:WorkflowViewStateService.ViewState>
                                      <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke Sub-Process #3" sap:VirtualizedContainerService.HintSize="372,87" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_29" UnSafe="False" WorkflowFileName="Sub-Processes\URNP - Extract Sentiment Analysis.xaml">
                                        <ui:InvokeWorkflowFile.Arguments>
                                          <InArgument x:TypeArguments="x:String" x:Key="in_StrAPIName">[in_Config("NLP_APIName").ToString]</InArgument>
                                          <InArgument x:TypeArguments="x:String" x:Key="in_StrKey">[in_Config("NLP_APIKey").ToString]</InArgument>
                                          <InArgument x:TypeArguments="x:String" x:Key="in_StrServiceUrl">[in_Config("NLP_SreviceURL").ToString]</InArgument>
                                          <InArgument x:TypeArguments="x:String" x:Key="in_StrUsername">[in_Config("NPL_UserName").ToString]</InArgument>
                                          <InArgument x:TypeArguments="x:String" x:Key="in_StrPassword">[in_Config("NPL_Password").ToString]</InArgument>
                                          <InArgument x:TypeArguments="x:String" x:Key="in_StrServiceVersion">[in_Config("NPL_ServiceVersion").ToString]</InArgument>
                                          <InArgument x:TypeArguments="x:String" x:Key="in_StrLanguage">[in_Config("NPL_Language").ToString]</InArgument>
                                          <InOutArgument x:TypeArguments="sd:DataTable" x:Key="io_DtNLPProcessDataTable">[dtNLPDataReviews]</InOutArgument>
                                        </ui:InvokeWorkflowFile.Arguments>
                                      </ui:InvokeWorkflowFile>
                                    </Sequence>
                                  </TryCatch.Try>
                                  <TryCatch.Catches>
                                    <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="418,21" sap2010:WorkflowViewState.IdRef="Catch`1_23">
                                      <sap:WorkflowViewStateService.ViewState>
                                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                          <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                          <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                        </scg:Dictionary>
                                      </sap:WorkflowViewStateService.ViewState>
                                      <ActivityAction x:TypeArguments="s:Exception">
                                        <ActivityAction.Argument>
                                          <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                        </ActivityAction.Argument>
                                        <Sequence DisplayName="Setup New Exception (Amalgamating Message/Source) Then Throw" sap:VirtualizedContainerService.HintSize="304,211" sap2010:WorkflowViewState.IdRef="Sequence_23">
                                          <sap:WorkflowViewStateService.ViewState>
                                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                            </scg:Dictionary>
                                          </sap:WorkflowViewStateService.ViewState>
                                          <Assign DisplayName="Assign ProcessApplicationException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_47">
                                            <Assign.To>
                                              <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                                            </Assign.To>
                                            <Assign.Value>
                                              <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                                            </Assign.Value>
                                          </Assign>
                                          <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="262,22" sap2010:WorkflowViewState.IdRef="Throw_19" />
                                        </Sequence>
                                      </ActivityAction>
                                    </Catch>
                                    <Catch x:TypeArguments="ui:BusinessRuleException" sap:VirtualizedContainerService.HintSize="418,21" sap2010:WorkflowViewState.IdRef="Catch`1_24">
                                      <sap:WorkflowViewStateService.ViewState>
                                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                          <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                          <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                        </scg:Dictionary>
                                      </sap:WorkflowViewStateService.ViewState>
                                      <ActivityAction x:TypeArguments="ui:BusinessRuleException">
                                        <ActivityAction.Argument>
                                          <DelegateInArgument x:TypeArguments="ui:BusinessRuleException" Name="exception" />
                                        </ActivityAction.Argument>
                                        <Sequence DisplayName="Setup New Exception (Amalgamating Message/Source) Then Throw" sap:VirtualizedContainerService.HintSize="304,211" sap2010:WorkflowViewState.IdRef="Sequence_24">
                                          <sap:WorkflowViewStateService.ViewState>
                                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                            </scg:Dictionary>
                                          </sap:WorkflowViewStateService.ViewState>
                                          <Assign DisplayName="Assign ProcessBusinessException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_48">
                                            <Assign.To>
                                              <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                                            </Assign.To>
                                            <Assign.Value>
                                              <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                                            </Assign.Value>
                                          </Assign>
                                          <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="262,22" sap2010:WorkflowViewState.IdRef="Throw_20" />
                                        </Sequence>
                                      </ActivityAction>
                                    </Catch>
                                  </TryCatch.Catches>
                                </TryCatch>
                                <FlowStep.Next>
                                  <FlowDecision x:Name="__ReferenceID22" Condition="[IsNothing(ProcessApplicationException) AND IsNothing(ProcessBusinessException)]" DisplayName="AE/BE Success?" sap:VirtualizedContainerService.HintSize="87,87" sap2010:WorkflowViewState.IdRef="FlowDecision_10">
                                    <sap:WorkflowViewStateService.ViewState>
                                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                        <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                        <av:Point x:Key="ShapeLocation">275,1549</av:Point>
                                        <av:Size x:Key="ShapeSize">87,87</av:Size>
                                        <av:PointCollection x:Key="FalseConnector">362,1592.5 392,1592.5 392,1546.5 730,1546.5 730,1576.5</av:PointCollection>
                                        <av:PointCollection x:Key="TrueConnector">275,1592.5 250,1592.5 250,1691 310,1691 310,1721</av:PointCollection>
                                      </scg:Dictionary>
                                    </sap:WorkflowViewStateService.ViewState>
                                    <FlowDecision.True>
                                      <FlowStep x:Name="__ReferenceID28">
                                        <sap:WorkflowViewStateService.ViewState>
                                          <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                            <av:Point x:Key="ShapeLocation">210,1721</av:Point>
                                            <av:Size x:Key="ShapeSize">200,78</av:Size>
                                            <av:PointCollection x:Key="ConnectorLocation">310,1799 310,1829 318.5,1829 318.5,1849</av:PointCollection>
                                          </scg:Dictionary>
                                        </sap:WorkflowViewStateService.ViewState>
                                        <TryCatch sap2010:Annotation.AnnotationText="No AE/BE Handling Required" DisplayName="TryCatch SubProcess #6 Create Report" sap:VirtualizedContainerService.HintSize="200,78" sap2010:WorkflowViewState.IdRef="TryCatch_18">
                                          <sap:WorkflowViewStateService.ViewState>
                                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                              <x:Boolean x:Key="IsAnnotationDocked">True</x:Boolean>
                                            </scg:Dictionary>
                                          </sap:WorkflowViewStateService.ViewState>
                                          <TryCatch.Try>
                                            <Sequence DisplayName="Invoke Sub-Process #6 Create Report" sap:VirtualizedContainerService.HintSize="405,179" sap2010:WorkflowViewState.IdRef="Sequence_25">
                                              <sap:WorkflowViewStateService.ViewState>
                                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                  <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                                </scg:Dictionary>
                                              </sap:WorkflowViewStateService.ViewState>
                                              <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke Sub-Process #3" sap:VirtualizedContainerService.HintSize="363,87" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_32" UnSafe="False" WorkflowFileName="Sub-Processes\URNP - Create Report And Reply.xaml">
                                                <ui:InvokeWorkflowFile.Arguments>
                                                  <InArgument x:TypeArguments="sd:DataTable" x:Key="in_DtNLPProcessDataTable">[dtNLPDataReviews]</InArgument>
                                                  <InArgument x:TypeArguments="x:String" x:Key="in_StrProductName">[strProductName]</InArgument>
                                                  <InArgument x:TypeArguments="snm:MailMessage" x:Key="in_MmsgEmailMessage">[in_TransactionItem]</InArgument>
                                                  <InArgument x:TypeArguments="scg:Dictionary(x:String, x:Int32)" x:Key="in_DictAmazonProductReview">[dictAmazonProductReview]</InArgument>
                                                  <InArgument x:TypeArguments="scg:Dictionary(x:String, x:String)" x:Key="in_DictFlipkartProductReview">[dictFlipkartProductReview]</InArgument>
                                                  <InArgument x:TypeArguments="scg:Dictionary(x:String, x:String)" x:Key="in_DictGoogleProductReview">[dictGoogleProductReview]</InArgument>
                                                  <InArgument x:TypeArguments="x:String" x:Key="in_StrNLPBadThreshold">[in_Config("NLPBadThreshold").ToString]</InArgument>
                                                  <InArgument x:TypeArguments="x:String" x:Key="in_StrNLPAvarageThreshold">[in_Config("NLPAverageThreshold").ToString]</InArgument>
                                                  <InArgument x:TypeArguments="x:String" x:Key="in_StrHTMLFilePath">[in_Config("HTMLTemplate_FilePath").ToString]</InArgument>
                                                </ui:InvokeWorkflowFile.Arguments>
                                              </ui:InvokeWorkflowFile>
                                            </Sequence>
                                          </TryCatch.Try>
                                          <TryCatch.Catches>
                                            <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="409,21" sap2010:WorkflowViewState.IdRef="Catch`1_27">
                                              <sap:WorkflowViewStateService.ViewState>
                                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                  <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                                </scg:Dictionary>
                                              </sap:WorkflowViewStateService.ViewState>
                                              <ActivityAction x:TypeArguments="s:Exception">
                                                <ActivityAction.Argument>
                                                  <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                                </ActivityAction.Argument>
                                                <Sequence DisplayName="Setup New Exception (Amalgamating Message/Source) Then Throw" sap:VirtualizedContainerService.HintSize="304,211" sap2010:WorkflowViewState.IdRef="Sequence_26">
                                                  <sap:WorkflowViewStateService.ViewState>
                                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                      <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                    </scg:Dictionary>
                                                  </sap:WorkflowViewStateService.ViewState>
                                                  <Assign DisplayName="Assign ProcessApplicationException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_51">
                                                    <Assign.To>
                                                      <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                                                    </Assign.To>
                                                    <Assign.Value>
                                                      <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                                                    </Assign.Value>
                                                  </Assign>
                                                  <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="262,22" sap2010:WorkflowViewState.IdRef="Throw_23" />
                                                </Sequence>
                                              </ActivityAction>
                                            </Catch>
                                            <Catch x:TypeArguments="ui:BusinessRuleException" sap:VirtualizedContainerService.HintSize="409,21" sap2010:WorkflowViewState.IdRef="Catch`1_28">
                                              <sap:WorkflowViewStateService.ViewState>
                                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                  <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                                </scg:Dictionary>
                                              </sap:WorkflowViewStateService.ViewState>
                                              <ActivityAction x:TypeArguments="ui:BusinessRuleException">
                                                <ActivityAction.Argument>
                                                  <DelegateInArgument x:TypeArguments="ui:BusinessRuleException" Name="exception" />
                                                </ActivityAction.Argument>
                                                <Sequence DisplayName="Setup New Exception (Amalgamating Message/Source) Then Throw" sap:VirtualizedContainerService.HintSize="304,211" sap2010:WorkflowViewState.IdRef="Sequence_27">
                                                  <sap:WorkflowViewStateService.ViewState>
                                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                      <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                    </scg:Dictionary>
                                                  </sap:WorkflowViewStateService.ViewState>
                                                  <Assign DisplayName="Assign ProcessBusinessException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_52">
                                                    <Assign.To>
                                                      <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                                                    </Assign.To>
                                                    <Assign.Value>
                                                      <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                                                    </Assign.Value>
                                                  </Assign>
                                                  <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="262,22" sap2010:WorkflowViewState.IdRef="Throw_24" />
                                                </Sequence>
                                              </ActivityAction>
                                            </Catch>
                                          </TryCatch.Catches>
                                        </TryCatch>
                                        <FlowStep.Next>
                                          <FlowDecision x:Name="__ReferenceID27" Condition="[IsNothing(ProcessApplicationException) AND IsNothing(ProcessBusinessException)]" DisplayName="AE/BE Success?" sap:VirtualizedContainerService.HintSize="87,87" sap2010:WorkflowViewState.IdRef="FlowDecision_12">
                                            <sap:WorkflowViewStateService.ViewState>
                                              <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                <av:Point x:Key="ShapeLocation">275,1849</av:Point>
                                                <av:Size x:Key="ShapeSize">87,87</av:Size>
                                                <av:PointCollection x:Key="FalseConnector">362,1892.5 392,1892.5 392,1846.5 730,1846.5 730,1876.5</av:PointCollection>
                                                <av:PointCollection x:Key="TrueConnector">305,810 280,810 280,931 340,931 340,961</av:PointCollection>
                                              </scg:Dictionary>
                                            </sap:WorkflowViewStateService.ViewState>
                                            <FlowDecision.False>
                                              <FlowDecision x:Name="__ReferenceID25" Condition="[IsNothing(ProcessBusinessException)]" DisplayName="AE?" sap:VirtualizedContainerService.HintSize="70,87" sap2010:WorkflowViewState.IdRef="FlowDecision_11">
                                                <sap:WorkflowViewStateService.ViewState>
                                                  <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                    <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                    <av:Point x:Key="ShapeLocation">695,1876.5</av:Point>
                                                    <av:Size x:Key="ShapeSize">70,87</av:Size>
                                                    <av:PointCollection x:Key="TrueConnector">695,1920 620,1920 620,1994.5</av:PointCollection>
                                                    <av:PointCollection x:Key="FalseConnector">765,1920 855,1920 855,1994.5</av:PointCollection>
                                                  </scg:Dictionary>
                                                </sap:WorkflowViewStateService.ViewState>
                                                <FlowDecision.True>
                                                  <FlowStep x:Name="__ReferenceID26">
                                                    <sap:WorkflowViewStateService.ViewState>
                                                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                        <av:Point x:Key="ShapeLocation">520,1994.5</av:Point>
                                                        <av:Size x:Key="ShapeSize">200,51</av:Size>
                                                      </scg:Dictionary>
                                                    </sap:WorkflowViewStateService.ViewState>
                                                    <TryCatch DisplayName="TryCatch AEHandler #6" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_19">
                                                      <TryCatch.Variables>
                                                        <Variable x:TypeArguments="x:String" Name="strOriginalExceptionMessage" />
                                                        <Variable x:TypeArguments="x:String" Name="strOriginalExceptionSource" />
                                                      </TryCatch.Variables>
                                                      <sap:WorkflowViewStateService.ViewState>
                                                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                        </scg:Dictionary>
                                                      </sap:WorkflowViewStateService.ViewState>
                                                      <TryCatch.Try>
                                                        <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke AEHandler #1" sap:VirtualizedContainerService.HintSize="334,89" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_33" UnSafe="False">
                                                          <ui:InvokeWorkflowFile.Arguments>
                                                            <scg:Dictionary x:TypeArguments="x:String, Argument" />
                                                          </ui:InvokeWorkflowFile.Arguments>
                                                          <ui:InvokeWorkflowFile.WorkflowFileName>
                                                            <InArgument x:TypeArguments="x:String">
                                                              <Literal x:TypeArguments="x:String" Value="" />
                                                            </InArgument>
                                                          </ui:InvokeWorkflowFile.WorkflowFileName>
                                                        </ui:InvokeWorkflowFile>
                                                      </TryCatch.Try>
                                                      <TryCatch.Catches>
                                                        <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="404,21" sap2010:WorkflowViewState.IdRef="Catch`1_29">
                                                          <sap:WorkflowViewStateService.ViewState>
                                                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                              <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                                              <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                                            </scg:Dictionary>
                                                          </sap:WorkflowViewStateService.ViewState>
                                                          <ActivityAction x:TypeArguments="s:Exception">
                                                            <ActivityAction.Argument>
                                                              <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                                            </ActivityAction.Argument>
                                                            <Assign DisplayName="Assign ProcessApplicationException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_53">
                                                              <Assign.To>
                                                                <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                                                              </Assign.To>
                                                              <Assign.Value>
                                                                <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessApplicationException.Message, exception.Message, exception.Source))]</InArgument>
                                                              </Assign.Value>
                                                            </Assign>
                                                          </ActivityAction>
                                                        </Catch>
                                                      </TryCatch.Catches>
                                                      <TryCatch.Finally>
                                                        <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_25" />
                                                      </TryCatch.Finally>
                                                    </TryCatch>
                                                  </FlowStep>
                                                </FlowDecision.True>
                                                <FlowDecision.False>
                                                  <FlowStep x:Name="__ReferenceID24">
                                                    <sap:WorkflowViewStateService.ViewState>
                                                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                        <av:Point x:Key="ShapeLocation">755,1994.5</av:Point>
                                                        <av:Size x:Key="ShapeSize">200,51</av:Size>
                                                      </scg:Dictionary>
                                                    </sap:WorkflowViewStateService.ViewState>
                                                    <TryCatch DisplayName="TryCatch BEHandler #6" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_20">
                                                      <sap:WorkflowViewStateService.ViewState>
                                                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                        </scg:Dictionary>
                                                      </sap:WorkflowViewStateService.ViewState>
                                                      <TryCatch.Try>
                                                        <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke BEHandler #1" sap:VirtualizedContainerService.HintSize="313.6,87.2" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_34" UnSafe="False">
                                                          <ui:InvokeWorkflowFile.Arguments>
                                                            <scg:Dictionary x:TypeArguments="x:String, Argument" />
                                                          </ui:InvokeWorkflowFile.Arguments>
                                                          <ui:InvokeWorkflowFile.WorkflowFileName>
                                                            <InArgument x:TypeArguments="x:String">
                                                              <Literal x:TypeArguments="x:String" Value="" />
                                                            </InArgument>
                                                          </ui:InvokeWorkflowFile.WorkflowFileName>
                                                        </ui:InvokeWorkflowFile>
                                                      </TryCatch.Try>
                                                      <TryCatch.Catches>
                                                        <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="400,130.4" sap2010:WorkflowViewState.IdRef="Catch`1_30">
                                                          <sap:WorkflowViewStateService.ViewState>
                                                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                              <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                                            </scg:Dictionary>
                                                          </sap:WorkflowViewStateService.ViewState>
                                                          <ActivityAction x:TypeArguments="s:Exception">
                                                            <ActivityAction.Argument>
                                                              <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                                            </ActivityAction.Argument>
                                                            <Assign DisplayName="Assign ProcessBusinessException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="243.2,60" sap2010:WorkflowViewState.IdRef="Assign_54">
                                                              <Assign.To>
                                                                <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                                                              </Assign.To>
                                                              <Assign.Value>
                                                                <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessBusinessException.Message, exception.Message, exception.Source))]</InArgument>
                                                              </Assign.Value>
                                                            </Assign>
                                                          </ActivityAction>
                                                        </Catch>
                                                      </TryCatch.Catches>
                                                      <TryCatch.Finally>
                                                        <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_26" />
                                                      </TryCatch.Finally>
                                                    </TryCatch>
                                                  </FlowStep>
                                                </FlowDecision.False>
                                              </FlowDecision>
                                            </FlowDecision.False>
                                          </FlowDecision>
                                        </FlowStep.Next>
                                      </FlowStep>
                                    </FlowDecision.True>
                                    <FlowDecision.False>
                                      <FlowDecision x:Name="__ReferenceID21" Condition="[IsNothing(ProcessBusinessException)]" DisplayName="AE?" sap:VirtualizedContainerService.HintSize="70,87" sap2010:WorkflowViewState.IdRef="FlowDecision_9">
                                        <sap:WorkflowViewStateService.ViewState>
                                          <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                            <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                            <av:Point x:Key="ShapeLocation">695,1576.5</av:Point>
                                            <av:Size x:Key="ShapeSize">70,87</av:Size>
                                            <av:PointCollection x:Key="TrueConnector">695,1620 620,1620 620,1694.5</av:PointCollection>
                                            <av:PointCollection x:Key="FalseConnector">765,1620 855,1620 855,1694.5</av:PointCollection>
                                          </scg:Dictionary>
                                        </sap:WorkflowViewStateService.ViewState>
                                        <FlowDecision.True>
                                          <FlowStep x:Name="__ReferenceID20">
                                            <sap:WorkflowViewStateService.ViewState>
                                              <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                <av:Point x:Key="ShapeLocation">520,1694.5</av:Point>
                                                <av:Size x:Key="ShapeSize">200,51</av:Size>
                                              </scg:Dictionary>
                                            </sap:WorkflowViewStateService.ViewState>
                                            <TryCatch DisplayName="TryCatch AEHandler #5" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_16">
                                              <TryCatch.Variables>
                                                <Variable x:TypeArguments="x:String" Name="strOriginalExceptionMessage" />
                                                <Variable x:TypeArguments="x:String" Name="strOriginalExceptionSource" />
                                              </TryCatch.Variables>
                                              <sap:WorkflowViewStateService.ViewState>
                                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                  <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                </scg:Dictionary>
                                              </sap:WorkflowViewStateService.ViewState>
                                              <TryCatch.Try>
                                                <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke AEHandler #1" sap:VirtualizedContainerService.HintSize="334,89" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_30" UnSafe="False">
                                                  <ui:InvokeWorkflowFile.Arguments>
                                                    <scg:Dictionary x:TypeArguments="x:String, Argument" />
                                                  </ui:InvokeWorkflowFile.Arguments>
                                                  <ui:InvokeWorkflowFile.WorkflowFileName>
                                                    <InArgument x:TypeArguments="x:String">
                                                      <Literal x:TypeArguments="x:String" Value="" />
                                                    </InArgument>
                                                  </ui:InvokeWorkflowFile.WorkflowFileName>
                                                </ui:InvokeWorkflowFile>
                                              </TryCatch.Try>
                                              <TryCatch.Catches>
                                                <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="404,21" sap2010:WorkflowViewState.IdRef="Catch`1_25">
                                                  <sap:WorkflowViewStateService.ViewState>
                                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                      <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                                      <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                                    </scg:Dictionary>
                                                  </sap:WorkflowViewStateService.ViewState>
                                                  <ActivityAction x:TypeArguments="s:Exception">
                                                    <ActivityAction.Argument>
                                                      <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                                    </ActivityAction.Argument>
                                                    <Assign DisplayName="Assign ProcessApplicationException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_49">
                                                      <Assign.To>
                                                        <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                                                      </Assign.To>
                                                      <Assign.Value>
                                                        <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessApplicationException.Message, exception.Message, exception.Source))]</InArgument>
                                                      </Assign.Value>
                                                    </Assign>
                                                  </ActivityAction>
                                                </Catch>
                                              </TryCatch.Catches>
                                              <TryCatch.Finally>
                                                <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_21" />
                                              </TryCatch.Finally>
                                            </TryCatch>
                                          </FlowStep>
                                        </FlowDecision.True>
                                        <FlowDecision.False>
                                          <FlowStep x:Name="__ReferenceID19">
                                            <sap:WorkflowViewStateService.ViewState>
                                              <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                <av:Point x:Key="ShapeLocation">755,1694.5</av:Point>
                                                <av:Size x:Key="ShapeSize">200,51</av:Size>
                                              </scg:Dictionary>
                                            </sap:WorkflowViewStateService.ViewState>
                                            <TryCatch DisplayName="TryCatch BEHandler #5" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_17">
                                              <sap:WorkflowViewStateService.ViewState>
                                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                  <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                </scg:Dictionary>
                                              </sap:WorkflowViewStateService.ViewState>
                                              <TryCatch.Try>
                                                <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke BEHandler #1" sap:VirtualizedContainerService.HintSize="313.6,87.2" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_31" UnSafe="False">
                                                  <ui:InvokeWorkflowFile.Arguments>
                                                    <scg:Dictionary x:TypeArguments="x:String, Argument" />
                                                  </ui:InvokeWorkflowFile.Arguments>
                                                  <ui:InvokeWorkflowFile.WorkflowFileName>
                                                    <InArgument x:TypeArguments="x:String">
                                                      <Literal x:TypeArguments="x:String" Value="" />
                                                    </InArgument>
                                                  </ui:InvokeWorkflowFile.WorkflowFileName>
                                                </ui:InvokeWorkflowFile>
                                              </TryCatch.Try>
                                              <TryCatch.Catches>
                                                <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="400,130.4" sap2010:WorkflowViewState.IdRef="Catch`1_26">
                                                  <sap:WorkflowViewStateService.ViewState>
                                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                                      <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                                      <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                                    </scg:Dictionary>
                                                  </sap:WorkflowViewStateService.ViewState>
                                                  <ActivityAction x:TypeArguments="s:Exception">
                                                    <ActivityAction.Argument>
                                                      <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                                    </ActivityAction.Argument>
                                                    <Assign DisplayName="Assign ProcessBusinessException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="243.2,60" sap2010:WorkflowViewState.IdRef="Assign_50">
                                                      <Assign.To>
                                                        <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                                                      </Assign.To>
                                                      <Assign.Value>
                                                        <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessBusinessException.Message, exception.Message, exception.Source))]</InArgument>
                                                      </Assign.Value>
                                                    </Assign>
                                                  </ActivityAction>
                                                </Catch>
                                              </TryCatch.Catches>
                                              <TryCatch.Finally>
                                                <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_22" />
                                              </TryCatch.Finally>
                                            </TryCatch>
                                          </FlowStep>
                                        </FlowDecision.False>
                                      </FlowDecision>
                                    </FlowDecision.False>
                                  </FlowDecision>
                                </FlowStep.Next>
                              </FlowStep>
                            </FlowDecision.True>
                            <FlowDecision.False>
                              <FlowDecision x:Name="__ReferenceID11" Condition="[IsNothing(ProcessBusinessException)]" DisplayName="AE?" sap:VirtualizedContainerService.HintSize="70,87" sap2010:WorkflowViewState.IdRef="FlowDecision_6">
                                <sap:WorkflowViewStateService.ViewState>
                                  <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                    <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                    <av:Point x:Key="ShapeLocation">695,1276.5</av:Point>
                                    <av:Size x:Key="ShapeSize">70,87</av:Size>
                                    <av:PointCollection x:Key="TrueConnector">695,1320 620,1320 620,1394.5</av:PointCollection>
                                    <av:PointCollection x:Key="FalseConnector">765,1320 855,1320 855,1394.5</av:PointCollection>
                                  </scg:Dictionary>
                                </sap:WorkflowViewStateService.ViewState>
                                <FlowDecision.True>
                                  <FlowStep x:Name="__ReferenceID12">
                                    <sap:WorkflowViewStateService.ViewState>
                                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                        <av:Point x:Key="ShapeLocation">520,1394.5</av:Point>
                                        <av:Size x:Key="ShapeSize">200,51</av:Size>
                                      </scg:Dictionary>
                                    </sap:WorkflowViewStateService.ViewState>
                                    <TryCatch DisplayName="TryCatch AEHandler #4" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_10">
                                      <TryCatch.Variables>
                                        <Variable x:TypeArguments="x:String" Name="strOriginalExceptionMessage" />
                                        <Variable x:TypeArguments="x:String" Name="strOriginalExceptionSource" />
                                      </TryCatch.Variables>
                                      <sap:WorkflowViewStateService.ViewState>
                                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                        </scg:Dictionary>
                                      </sap:WorkflowViewStateService.ViewState>
                                      <TryCatch.Try>
                                        <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke AEHandler #1" sap:VirtualizedContainerService.HintSize="334,89" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_24" UnSafe="False">
                                          <ui:InvokeWorkflowFile.Arguments>
                                            <scg:Dictionary x:TypeArguments="x:String, Argument" />
                                          </ui:InvokeWorkflowFile.Arguments>
                                          <ui:InvokeWorkflowFile.WorkflowFileName>
                                            <InArgument x:TypeArguments="x:String">
                                              <Literal x:TypeArguments="x:String" Value="" />
                                            </InArgument>
                                          </ui:InvokeWorkflowFile.WorkflowFileName>
                                        </ui:InvokeWorkflowFile>
                                      </TryCatch.Try>
                                      <TryCatch.Catches>
                                        <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="404,21" sap2010:WorkflowViewState.IdRef="Catch`1_17">
                                          <sap:WorkflowViewStateService.ViewState>
                                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                              <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                              <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                            </scg:Dictionary>
                                          </sap:WorkflowViewStateService.ViewState>
                                          <ActivityAction x:TypeArguments="s:Exception">
                                            <ActivityAction.Argument>
                                              <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                            </ActivityAction.Argument>
                                            <Assign DisplayName="Assign ProcessApplicationException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_41">
                                              <Assign.To>
                                                <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                                              </Assign.To>
                                              <Assign.Value>
                                                <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessApplicationException.Message, exception.Message, exception.Source))]</InArgument>
                                              </Assign.Value>
                                            </Assign>
                                          </ActivityAction>
                                        </Catch>
                                      </TryCatch.Catches>
                                      <TryCatch.Finally>
                                        <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_15" />
                                      </TryCatch.Finally>
                                    </TryCatch>
                                  </FlowStep>
                                </FlowDecision.True>
                                <FlowDecision.False>
                                  <FlowStep x:Name="__ReferenceID13">
                                    <sap:WorkflowViewStateService.ViewState>
                                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                        <av:Point x:Key="ShapeLocation">755,1394.5</av:Point>
                                        <av:Size x:Key="ShapeSize">200,51</av:Size>
                                      </scg:Dictionary>
                                    </sap:WorkflowViewStateService.ViewState>
                                    <TryCatch DisplayName="TryCatch BEHandler #4" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_11">
                                      <sap:WorkflowViewStateService.ViewState>
                                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                        </scg:Dictionary>
                                      </sap:WorkflowViewStateService.ViewState>
                                      <TryCatch.Try>
                                        <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke BEHandler #1" sap:VirtualizedContainerService.HintSize="313.6,87.2" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_25" UnSafe="False">
                                          <ui:InvokeWorkflowFile.Arguments>
                                            <scg:Dictionary x:TypeArguments="x:String, Argument" />
                                          </ui:InvokeWorkflowFile.Arguments>
                                          <ui:InvokeWorkflowFile.WorkflowFileName>
                                            <InArgument x:TypeArguments="x:String">
                                              <Literal x:TypeArguments="x:String" Value="" />
                                            </InArgument>
                                          </ui:InvokeWorkflowFile.WorkflowFileName>
                                        </ui:InvokeWorkflowFile>
                                      </TryCatch.Try>
                                      <TryCatch.Catches>
                                        <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="400,130.4" sap2010:WorkflowViewState.IdRef="Catch`1_18">
                                          <sap:WorkflowViewStateService.ViewState>
                                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                              <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                            </scg:Dictionary>
                                          </sap:WorkflowViewStateService.ViewState>
                                          <ActivityAction x:TypeArguments="s:Exception">
                                            <ActivityAction.Argument>
                                              <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                            </ActivityAction.Argument>
                                            <Assign DisplayName="Assign ProcessBusinessException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="243.2,60" sap2010:WorkflowViewState.IdRef="Assign_42">
                                              <Assign.To>
                                                <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                                              </Assign.To>
                                              <Assign.Value>
                                                <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessBusinessException.Message, exception.Message, exception.Source))]</InArgument>
                                              </Assign.Value>
                                            </Assign>
                                          </ActivityAction>
                                        </Catch>
                                      </TryCatch.Catches>
                                      <TryCatch.Finally>
                                        <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_16" />
                                      </TryCatch.Finally>
                                    </TryCatch>
                                  </FlowStep>
                                </FlowDecision.False>
                              </FlowDecision>
                            </FlowDecision.False>
                          </FlowDecision>
                        </FlowStep.Next>
                      </FlowStep>
                    </FlowDecision.True>
                    <FlowDecision.False>
                      <FlowDecision x:Name="__ReferenceID8" Condition="[IsNothing(ProcessBusinessException)]" DisplayName="AE?" sap:VirtualizedContainerService.HintSize="70,87" sap2010:WorkflowViewState.IdRef="FlowDecision_4">
                        <sap:WorkflowViewStateService.ViewState>
                          <scg:Dictionary x:TypeArguments="x:String, x:Object">
                            <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                            <av:Point x:Key="ShapeLocation">685,956.5</av:Point>
                            <av:Size x:Key="ShapeSize">70,87</av:Size>
                            <av:PointCollection x:Key="TrueConnector">685,1000 610,1000 610,1094.5</av:PointCollection>
                            <av:PointCollection x:Key="FalseConnector">755,1000 830,1000 830,1094.5</av:PointCollection>
                          </scg:Dictionary>
                        </sap:WorkflowViewStateService.ViewState>
                        <FlowDecision.True>
                          <FlowStep x:Name="__ReferenceID6" sap2010:WorkflowViewState.IdRef="FlowStep_6">
                            <sap:WorkflowViewStateService.ViewState>
                              <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                <av:Point x:Key="ShapeLocation">510,1094.5</av:Point>
                                <av:Size x:Key="ShapeSize">200,51</av:Size>
                              </scg:Dictionary>
                            </sap:WorkflowViewStateService.ViewState>
                            <TryCatch DisplayName="TryCatch AEHandler #3" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_6">
                              <TryCatch.Variables>
                                <Variable x:TypeArguments="x:String" Name="strOriginalExceptionMessage" />
                                <Variable x:TypeArguments="x:String" Name="strOriginalExceptionSource" />
                              </TryCatch.Variables>
                              <sap:WorkflowViewStateService.ViewState>
                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                  <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                </scg:Dictionary>
                              </sap:WorkflowViewStateService.ViewState>
                              <TryCatch.Try>
                                <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke AEHandler #1" sap:VirtualizedContainerService.HintSize="314,87" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_15" UnSafe="False">
                                  <ui:InvokeWorkflowFile.Arguments>
                                    <scg:Dictionary x:TypeArguments="x:String, Argument" />
                                  </ui:InvokeWorkflowFile.Arguments>
                                  <ui:InvokeWorkflowFile.WorkflowFileName>
                                    <InArgument x:TypeArguments="x:String">
                                      <Literal x:TypeArguments="x:String" Value="" />
                                    </InArgument>
                                  </ui:InvokeWorkflowFile.WorkflowFileName>
                                </ui:InvokeWorkflowFile>
                              </TryCatch.Try>
                              <TryCatch.Catches>
                                <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="400,21" sap2010:WorkflowViewState.IdRef="Catch`1_11">
                                  <sap:WorkflowViewStateService.ViewState>
                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                      <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                                      <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                    </scg:Dictionary>
                                  </sap:WorkflowViewStateService.ViewState>
                                  <ActivityAction x:TypeArguments="s:Exception">
                                    <ActivityAction.Argument>
                                      <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                    </ActivityAction.Argument>
                                    <Assign DisplayName="Assign ProcessApplicationException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_36">
                                      <Assign.To>
                                        <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                                      </Assign.To>
                                      <Assign.Value>
                                        <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessApplicationException.Message, exception.Message, exception.Source))]</InArgument>
                                      </Assign.Value>
                                    </Assign>
                                  </ActivityAction>
                                </Catch>
                              </TryCatch.Catches>
                              <TryCatch.Finally>
                                <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_9" />
                              </TryCatch.Finally>
                            </TryCatch>
                          </FlowStep>
                        </FlowDecision.True>
                        <FlowDecision.False>
                          <FlowStep x:Name="__ReferenceID9">
                            <sap:WorkflowViewStateService.ViewState>
                              <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                <av:Point x:Key="ShapeLocation">730,1094.5</av:Point>
                                <av:Size x:Key="ShapeSize">200,51</av:Size>
                              </scg:Dictionary>
                            </sap:WorkflowViewStateService.ViewState>
                            <TryCatch DisplayName="TryCatch BEHandler #3" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_8">
                              <sap:WorkflowViewStateService.ViewState>
                                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                  <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                </scg:Dictionary>
                              </sap:WorkflowViewStateService.ViewState>
                              <TryCatch.Try>
                                <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke BEHandler #1" sap:VirtualizedContainerService.HintSize="313.6,87.2" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_22" UnSafe="False">
                                  <ui:InvokeWorkflowFile.Arguments>
                                    <scg:Dictionary x:TypeArguments="x:String, Argument" />
                                  </ui:InvokeWorkflowFile.Arguments>
                                  <ui:InvokeWorkflowFile.WorkflowFileName>
                                    <InArgument x:TypeArguments="x:String">
                                      <Literal x:TypeArguments="x:String" Value="" />
                                    </InArgument>
                                  </ui:InvokeWorkflowFile.WorkflowFileName>
                                </ui:InvokeWorkflowFile>
                              </TryCatch.Try>
                              <TryCatch.Catches>
                                <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="400,130.4" sap2010:WorkflowViewState.IdRef="Catch`1_14">
                                  <sap:WorkflowViewStateService.ViewState>
                                    <scg:Dictionary x:TypeArguments="x:String, x:Object">
                                      <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                                      <x:Boolean x:Key="IsPinned">False</x:Boolean>
                                    </scg:Dictionary>
                                  </sap:WorkflowViewStateService.ViewState>
                                  <ActivityAction x:TypeArguments="s:Exception">
                                    <ActivityAction.Argument>
                                      <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                                    </ActivityAction.Argument>
                                    <Assign DisplayName="Assign ProcessBusinessException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="243.2,60" sap2010:WorkflowViewState.IdRef="Assign_38">
                                      <Assign.To>
                                        <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                                      </Assign.To>
                                      <Assign.Value>
                                        <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessBusinessException.Message, exception.Message, exception.Source))]</InArgument>
                                      </Assign.Value>
                                    </Assign>
                                  </ActivityAction>
                                </Catch>
                              </TryCatch.Catches>
                              <TryCatch.Finally>
                                <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_13" />
                              </TryCatch.Finally>
                            </TryCatch>
                          </FlowStep>
                        </FlowDecision.False>
                      </FlowDecision>
                    </FlowDecision.False>
                  </FlowDecision>
                </FlowStep.Next>
              </FlowStep>
            </FlowDecision.True>
            <FlowDecision.False>
              <FlowDecision x:Name="__ReferenceID4" Condition="[IsNothing(ProcessBusinessException)]" DisplayName="AE?" sap:VirtualizedContainerService.HintSize="70,87" sap2010:WorkflowViewState.IdRef="FlowDecision_3">
                <sap:WorkflowViewStateService.ViewState>
                  <scg:Dictionary x:TypeArguments="x:String, x:Object">
                    <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                    <av:Point x:Key="ShapeLocation">685,606.5</av:Point>
                    <av:Size x:Key="ShapeSize">70,87</av:Size>
                    <av:PointCollection x:Key="TrueConnector">685,650 610,650 610,744.5</av:PointCollection>
                    <av:PointCollection x:Key="FalseConnector">755,650 840,650 840,744.5</av:PointCollection>
                  </scg:Dictionary>
                </sap:WorkflowViewStateService.ViewState>
                <FlowDecision.True>
                  <FlowStep x:Name="__ReferenceID1" sap2010:WorkflowViewState.IdRef="FlowStep_2">
                    <sap:WorkflowViewStateService.ViewState>
                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                        <av:Point x:Key="ShapeLocation">510,744.5</av:Point>
                        <av:Size x:Key="ShapeSize">200,51</av:Size>
                      </scg:Dictionary>
                    </sap:WorkflowViewStateService.ViewState>
                    <TryCatch DisplayName="TryCatch AEHandler #1" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_2">
                      <sap:WorkflowViewStateService.ViewState>
                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                        </scg:Dictionary>
                      </sap:WorkflowViewStateService.ViewState>
                      <TryCatch.Try>
                        <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke AEHandler #1" sap:VirtualizedContainerService.HintSize="313.6,87.2" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_20" UnSafe="False">
                          <ui:InvokeWorkflowFile.Arguments>
                            <scg:Dictionary x:TypeArguments="x:String, Argument" />
                          </ui:InvokeWorkflowFile.Arguments>
                          <ui:InvokeWorkflowFile.WorkflowFileName>
                            <InArgument x:TypeArguments="x:String">
                              <Literal x:TypeArguments="x:String" Value="" />
                            </InArgument>
                          </ui:InvokeWorkflowFile.WorkflowFileName>
                        </ui:InvokeWorkflowFile>
                      </TryCatch.Try>
                      <TryCatch.Catches>
                        <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="400,130.4" sap2010:WorkflowViewState.IdRef="Catch`1_3">
                          <sap:WorkflowViewStateService.ViewState>
                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                              <x:Boolean x:Key="IsPinned">False</x:Boolean>
                            </scg:Dictionary>
                          </sap:WorkflowViewStateService.ViewState>
                          <ActivityAction x:TypeArguments="s:Exception">
                            <ActivityAction.Argument>
                              <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                            </ActivityAction.Argument>
                            <Assign DisplayName="Assign ProcessApplicationException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="243.2,60" sap2010:WorkflowViewState.IdRef="Assign_37">
                              <Assign.To>
                                <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                              </Assign.To>
                              <Assign.Value>
                                <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessBusinessException.Message, exception.Message, exception.Source))]</InArgument>
                              </Assign.Value>
                            </Assign>
                          </ActivityAction>
                        </Catch>
                      </TryCatch.Catches>
                      <TryCatch.Finally>
                        <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_7" />
                      </TryCatch.Finally>
                    </TryCatch>
                  </FlowStep>
                </FlowDecision.True>
                <FlowDecision.False>
                  <FlowStep x:Name="__ReferenceID5" sap2010:WorkflowViewState.IdRef="FlowStep_5">
                    <sap:WorkflowViewStateService.ViewState>
                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                        <av:Point x:Key="ShapeLocation">740,744.5</av:Point>
                        <av:Size x:Key="ShapeSize">200,51</av:Size>
                      </scg:Dictionary>
                    </sap:WorkflowViewStateService.ViewState>
                    <TryCatch DisplayName="TryCatch BEHandler #1" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_5">
                      <sap:WorkflowViewStateService.ViewState>
                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                        </scg:Dictionary>
                      </sap:WorkflowViewStateService.ViewState>
                      <TryCatch.Try>
                        <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke BEHandler #1" sap:VirtualizedContainerService.HintSize="313.6,87.2" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_14" UnSafe="False">
                          <ui:InvokeWorkflowFile.Arguments>
                            <scg:Dictionary x:TypeArguments="x:String, Argument" />
                          </ui:InvokeWorkflowFile.Arguments>
                          <ui:InvokeWorkflowFile.WorkflowFileName>
                            <InArgument x:TypeArguments="x:String">
                              <Literal x:TypeArguments="x:String" Value="" />
                            </InArgument>
                          </ui:InvokeWorkflowFile.WorkflowFileName>
                        </ui:InvokeWorkflowFile>
                      </TryCatch.Try>
                      <TryCatch.Catches>
                        <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="400,130.4" sap2010:WorkflowViewState.IdRef="Catch`1_9">
                          <sap:WorkflowViewStateService.ViewState>
                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                              <x:Boolean x:Key="IsPinned">False</x:Boolean>
                            </scg:Dictionary>
                          </sap:WorkflowViewStateService.ViewState>
                          <ActivityAction x:TypeArguments="s:Exception">
                            <ActivityAction.Argument>
                              <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                            </ActivityAction.Argument>
                            <Assign DisplayName="Assign ProcessBusinessException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="243.2,60" sap2010:WorkflowViewState.IdRef="Assign_29">
                              <Assign.To>
                                <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                              </Assign.To>
                              <Assign.Value>
                                <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessBusinessException.Message, exception.Message, exception.Source))]</InArgument>
                              </Assign.Value>
                            </Assign>
                          </ActivityAction>
                        </Catch>
                      </TryCatch.Catches>
                      <TryCatch.Finally>
                        <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_8" />
                      </TryCatch.Finally>
                    </TryCatch>
                  </FlowStep>
                </FlowDecision.False>
              </FlowDecision>
            </FlowDecision.False>
          </FlowDecision>
        </FlowStep.Next>
      </FlowStep>
      <x:Reference>__ReferenceID0</x:Reference>
      <x:Reference>__ReferenceID1</x:Reference>
      <x:Reference>__ReferenceID2</x:Reference>
      <x:Reference>__ReferenceID3</x:Reference>
      <x:Reference>__ReferenceID4</x:Reference>
      <x:Reference>__ReferenceID5</x:Reference>
      <x:Reference>__ReferenceID6</x:Reference>
      <x:Reference>__ReferenceID7</x:Reference>
      <x:Reference>__ReferenceID8</x:Reference>
      <x:Reference>__ReferenceID9</x:Reference>
      <x:Reference>__ReferenceID10</x:Reference>
      <x:Reference>__ReferenceID11</x:Reference>
      <x:Reference>__ReferenceID12</x:Reference>
      <x:Reference>__ReferenceID13</x:Reference>
      <FlowStep x:Name="__ReferenceID29">
        <sap:WorkflowViewStateService.ViewState>
          <scg:Dictionary x:TypeArguments="x:String, x:Object">
            <av:Point x:Key="ShapeLocation">200,121</av:Point>
            <av:Size x:Key="ShapeSize">200,78</av:Size>
            <av:PointCollection x:Key="ConnectorLocation">300,199 300,259</av:PointCollection>
          </scg:Dictionary>
        </sap:WorkflowViewStateService.ViewState>
        <TryCatch sap2010:Annotation.AnnotationText="AE/BE Handling Necessary" DisplayName="TryCatch SubProcess #1 Init Data" sap:VirtualizedContainerService.HintSize="459,455" sap2010:WorkflowViewState.IdRef="TryCatch_12">
          <sap:WorkflowViewStateService.ViewState>
            <scg:Dictionary x:TypeArguments="x:String, x:Object">
              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
              <x:Boolean x:Key="IsAnnotationDocked">True</x:Boolean>
            </scg:Dictionary>
          </sap:WorkflowViewStateService.ViewState>
          <TryCatch.Try>
            <Sequence DisplayName="Invoke Sub-Process #1 And Set Progress" sap:VirtualizedContainerService.HintSize="421,179" sap2010:WorkflowViewState.IdRef="Sequence_21">
              <sap:WorkflowViewStateService.ViewState>
                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                  <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                </scg:Dictionary>
              </sap:WorkflowViewStateService.ViewState>
              <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke Sub-Process #1 CPA - Get Product Details From Amazon" sap:VirtualizedContainerService.HintSize="379,87" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_26" UnSafe="False" WorkflowFileName="Sub-Processes/URNP - Application Data Initialization.xaml">
                <ui:InvokeWorkflowFile.Arguments>
                  <InOutArgument x:TypeArguments="sd:DataTable" x:Key="io_DtNLPProcessDataTable">[dtNLPDataReviews]</InOutArgument>
                  <InArgument x:TypeArguments="x:String" x:Key="in_strEmailBody">[in_TransactionItem.Body.ToString]</InArgument>
                  <OutArgument x:TypeArguments="x:String" x:Key="out_strProductName">[strProductName]</OutArgument>
                </ui:InvokeWorkflowFile.Arguments>
              </ui:InvokeWorkflowFile>
            </Sequence>
          </TryCatch.Try>
          <TryCatch.Catches>
            <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="425,21" sap2010:WorkflowViewState.IdRef="Catch`1_19">
              <sap:WorkflowViewStateService.ViewState>
                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                  <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                </scg:Dictionary>
              </sap:WorkflowViewStateService.ViewState>
              <ActivityAction x:TypeArguments="s:Exception">
                <ActivityAction.Argument>
                  <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                </ActivityAction.Argument>
                <Assign DisplayName="Assign ProcessApplicationException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_43">
                  <Assign.To>
                    <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                  </Assign.To>
                  <Assign.Value>
                    <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                  </Assign.Value>
                </Assign>
              </ActivityAction>
            </Catch>
            <Catch x:TypeArguments="ui:BusinessRuleException" sap:VirtualizedContainerService.HintSize="425,21" sap2010:WorkflowViewState.IdRef="Catch`1_20">
              <sap:WorkflowViewStateService.ViewState>
                <scg:Dictionary x:TypeArguments="x:String, x:Object">
                  <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                  <x:Boolean x:Key="IsPinned">False</x:Boolean>
                </scg:Dictionary>
              </sap:WorkflowViewStateService.ViewState>
              <ActivityAction x:TypeArguments="ui:BusinessRuleException">
                <ActivityAction.Argument>
                  <DelegateInArgument x:TypeArguments="ui:BusinessRuleException" Name="exception" />
                </ActivityAction.Argument>
                <Assign DisplayName="Assign ProcessBusinessException" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_44">
                  <Assign.To>
                    <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                  </Assign.To>
                  <Assign.Value>
                    <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("Failure Message: {0}. Failure Source: {1}.", exception.Message, exception.Source))]</InArgument>
                  </Assign.Value>
                </Assign>
              </ActivityAction>
            </Catch>
          </TryCatch.Catches>
        </TryCatch>
        <FlowStep.Next>
          <FlowDecision x:Name="__ReferenceID15" Condition="[IsNothing(ProcessApplicationException) AND IsNothing(ProcessBusinessException)]" DisplayName="AE/BE Success?" sap:VirtualizedContainerService.HintSize="87,87" sap2010:WorkflowViewState.IdRef="FlowDecision_7" True="{x:Reference __ReferenceID14}">
            <sap:WorkflowViewStateService.ViewState>
              <scg:Dictionary x:TypeArguments="x:String, x:Object">
                <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                <av:Point x:Key="ShapeLocation">256.5,259</av:Point>
                <av:Size x:Key="ShapeSize">87,87</av:Size>
                <av:PointCollection x:Key="FalseConnector">343.5,302.5 373.5,302.5 373.5,246.5 720,246.5 720,276.5</av:PointCollection>
                <av:PointCollection x:Key="TrueConnector">256.5,302.5 240,302.5 240,424.5 300,424.5 300,454.5</av:PointCollection>
              </scg:Dictionary>
            </sap:WorkflowViewStateService.ViewState>
            <FlowDecision.False>
              <FlowDecision x:Name="__ReferenceID18" Condition="[IsNothing(ProcessBusinessException)]" DisplayName="AE?" sap:VirtualizedContainerService.HintSize="70,87" sap2010:WorkflowViewState.IdRef="FlowDecision_8">
                <sap:WorkflowViewStateService.ViewState>
                  <scg:Dictionary x:TypeArguments="x:String, x:Object">
                    <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                    <av:Point x:Key="ShapeLocation">685,276.5</av:Point>
                    <av:Size x:Key="ShapeSize">70,87</av:Size>
                    <av:PointCollection x:Key="TrueConnector">685,320 610,320 610,414.5</av:PointCollection>
                    <av:PointCollection x:Key="FalseConnector">755,320 840,320 840,414.5</av:PointCollection>
                  </scg:Dictionary>
                </sap:WorkflowViewStateService.ViewState>
                <FlowDecision.True>
                  <FlowStep x:Name="__ReferenceID17">
                    <sap:WorkflowViewStateService.ViewState>
                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                        <av:Point x:Key="ShapeLocation">510,414.5</av:Point>
                        <av:Size x:Key="ShapeSize">200,51</av:Size>
                      </scg:Dictionary>
                    </sap:WorkflowViewStateService.ViewState>
                    <TryCatch DisplayName="TryCatch AEHandler #1" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_13">
                      <sap:WorkflowViewStateService.ViewState>
                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                        </scg:Dictionary>
                      </sap:WorkflowViewStateService.ViewState>
                      <TryCatch.Try>
                        <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke AEHandler #1" sap:VirtualizedContainerService.HintSize="334,89" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_27" UnSafe="False">
                          <ui:InvokeWorkflowFile.Arguments>
                            <scg:Dictionary x:TypeArguments="x:String, Argument" />
                          </ui:InvokeWorkflowFile.Arguments>
                          <ui:InvokeWorkflowFile.WorkflowFileName>
                            <InArgument x:TypeArguments="x:String">
                              <Literal x:TypeArguments="x:String" Value="" />
                            </InArgument>
                          </ui:InvokeWorkflowFile.WorkflowFileName>
                        </ui:InvokeWorkflowFile>
                      </TryCatch.Try>
                      <TryCatch.Catches>
                        <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="404,21" sap2010:WorkflowViewState.IdRef="Catch`1_21">
                          <sap:WorkflowViewStateService.ViewState>
                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                              <x:Boolean x:Key="IsExpanded">False</x:Boolean>
                              <x:Boolean x:Key="IsPinned">False</x:Boolean>
                            </scg:Dictionary>
                          </sap:WorkflowViewStateService.ViewState>
                          <ActivityAction x:TypeArguments="s:Exception">
                            <ActivityAction.Argument>
                              <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                            </ActivityAction.Argument>
                            <Assign DisplayName="Assign ProcessApplicationException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="262,60" sap2010:WorkflowViewState.IdRef="Assign_45">
                              <Assign.To>
                                <OutArgument x:TypeArguments="s:Exception">[ProcessApplicationException]</OutArgument>
                              </Assign.To>
                              <Assign.Value>
                                <InArgument x:TypeArguments="s:Exception">[new Exception(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessBusinessException.Message, exception.Message, exception.Source))]</InArgument>
                              </Assign.Value>
                            </Assign>
                          </ActivityAction>
                        </Catch>
                      </TryCatch.Catches>
                      <TryCatch.Finally>
                        <Throw DisplayName="Throw AE" Exception="[ProcessApplicationException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_17" />
                      </TryCatch.Finally>
                    </TryCatch>
                  </FlowStep>
                </FlowDecision.True>
                <FlowDecision.False>
                  <FlowStep x:Name="__ReferenceID16">
                    <sap:WorkflowViewStateService.ViewState>
                      <scg:Dictionary x:TypeArguments="x:String, x:Object">
                        <av:Point x:Key="ShapeLocation">740,414.5</av:Point>
                        <av:Size x:Key="ShapeSize">200,51</av:Size>
                      </scg:Dictionary>
                    </sap:WorkflowViewStateService.ViewState>
                    <TryCatch DisplayName="TryCatch BEHandler #1" sap:VirtualizedContainerService.HintSize="200,51" sap2010:WorkflowViewState.IdRef="TryCatch_14">
                      <sap:WorkflowViewStateService.ViewState>
                        <scg:Dictionary x:TypeArguments="x:String, x:Object">
                          <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                        </scg:Dictionary>
                      </sap:WorkflowViewStateService.ViewState>
                      <TryCatch.Try>
                        <ui:InvokeWorkflowFile ContinueOnError="{x:Null}" DisplayName="Invoke BEHandler #1" sap:VirtualizedContainerService.HintSize="313.6,87.2" sap2010:WorkflowViewState.IdRef="InvokeWorkflowFile_28" UnSafe="False">
                          <ui:InvokeWorkflowFile.Arguments>
                            <scg:Dictionary x:TypeArguments="x:String, Argument" />
                          </ui:InvokeWorkflowFile.Arguments>
                          <ui:InvokeWorkflowFile.WorkflowFileName>
                            <InArgument x:TypeArguments="x:String">
                              <Literal x:TypeArguments="x:String" Value="" />
                            </InArgument>
                          </ui:InvokeWorkflowFile.WorkflowFileName>
                        </ui:InvokeWorkflowFile>
                      </TryCatch.Try>
                      <TryCatch.Catches>
                        <Catch x:TypeArguments="s:Exception" sap:VirtualizedContainerService.HintSize="400,130.4" sap2010:WorkflowViewState.IdRef="Catch`1_22">
                          <sap:WorkflowViewStateService.ViewState>
                            <scg:Dictionary x:TypeArguments="x:String, x:Object">
                              <x:Boolean x:Key="IsExpanded">True</x:Boolean>
                              <x:Boolean x:Key="IsPinned">False</x:Boolean>
                            </scg:Dictionary>
                          </sap:WorkflowViewStateService.ViewState>
                          <ActivityAction x:TypeArguments="s:Exception">
                            <ActivityAction.Argument>
                              <DelegateInArgument x:TypeArguments="s:Exception" Name="exception" />
                            </ActivityAction.Argument>
                            <Assign DisplayName="Assign ProcessBusinessException (Message/Source In Message For Both Original And Handler Exception)" sap:VirtualizedContainerService.HintSize="243.2,60" sap2010:WorkflowViewState.IdRef="Assign_46">
                              <Assign.To>
                                <OutArgument x:TypeArguments="ui:BusinessRuleException">[ProcessBusinessException]</OutArgument>
                              </Assign.To>
                              <Assign.Value>
                                <InArgument x:TypeArguments="ui:BusinessRuleException">[new BusinessRuleException(String.Format("{0} Critical Failure Message: {1}. Critical Failure Source: {2}.", ProcessBusinessException.Message, exception.Message, exception.Source))]</InArgument>
                              </Assign.Value>
                            </Assign>
                          </ActivityAction>
                        </Catch>
                      </TryCatch.Catches>
                      <TryCatch.Finally>
                        <Throw DisplayName="Throw BE" Exception="[ProcessBusinessException]" sap:VirtualizedContainerService.HintSize="200,22" sap2010:WorkflowViewState.IdRef="Throw_18" />
                      </TryCatch.Finally>
                    </TryCatch>
                  </FlowStep>
                </FlowDecision.False>
              </FlowDecision>
            </FlowDecision.False>
          </FlowDecision>
        </FlowStep.Next>
      </FlowStep>
      <x:Reference>__ReferenceID15</x:Reference>
      <x:Reference>__ReferenceID16</x:Reference>
      <x:Reference>__ReferenceID17</x:Reference>
      <x:Reference>__ReferenceID18</x:Reference>
      <x:Reference>__ReferenceID19</x:Reference>
      <x:Reference>__ReferenceID20</x:Reference>
      <x:Reference>__ReferenceID21</x:Reference>
      <x:Reference>__ReferenceID22</x:Reference>
      <x:Reference>__ReferenceID23</x:Reference>
      <x:Reference>__ReferenceID24</x:Reference>
      <x:Reference>__ReferenceID25</x:Reference>
      <x:Reference>__ReferenceID26</x:Reference>
      <x:Reference>__ReferenceID27</x:Reference>
      <x:Reference>__ReferenceID28</x:Reference>
    </Flowchart>
  </Sequence>
</Activity>